cmd/scollector/collectors/httpunit.go

```diff
@@ -46,16 +48,32 @@ func cHTTPUnit(plans *httpunit.Plans) (opentsdb.MultiDataPoint, error) {
 	unix_now := time.Now().Unix()
 	var md opentsdb.MultiDataPoint
 	for r := range ch {
+		if r == nil || r.Case == nil || r.Plan == nil {
+			slog.Errorf("httpunit: unexpected test plan on channel, the plan result, the case, and/or the plan result is missing.")
+			continue
+		}
 		tags := opentsdb.TagSet{
 			"protocol":     r.Case.URL.Scheme,
 			"ip":           r.Case.IP.String(),
 			"url_host":     r.Case.URL.Host,
 			"hc_test_case": r.Plan.Label,
 		}
+		metadata.AddMeta("", opentsdb.TagSet{"hc_test_case": r.Plan.Label}, "url", r.Case.URL.String(), false)
+		if r.Result == nil || r.Result.Result != nil {
```

Comment:
what is this logic? If the result is nil or the result's result is not nil? Seems odd to set error and not continue in some cases, which then unsets error? Seems odd.